New article: Merge Conflicts on Flexible Environments (#7013)
umbraco/UmbracoDocs: umbraco-cloud/troubleshooting/deployments/merge-conflicts-on-flexible-environments.md

---

> description: Merge conflicts can happen on a flexible environment when the same schema has been changed in both the source and target environment.

Reviewer: ```suggestion
description: Merge conflicts can happen when the same schema has been modified in both the flexible environment and the mainline environment it's connected to.
```

---

# Merge Conflicts on Flexible Environments

> A merge conflict happens when one file or item contains changes in more than one instance that's part of a deployment.

Reviewer: ```suggestion
A merge conflict happens when a file or item contains conflicting changes in more than one environment during deployment.
```

This guide explains how to resolve these merge conflicts and how to avoid them.

## How to Resolve a Merge Conflict on a Flexible Environment

The image below shows a merge conflict happening on a pull from the mainline environment to a flexible environment.

![Pulling changes from the mainline environment failed](images/pulling-failed.png)

To start debugging the merge conflict, a log file containing a list of conflicting files is provided. It's a `.txt` file that can be downloaded and viewed.

### Preparations

Before you start resolving the merge conflict, ensure you have the following things ready:

* The log file containing a list of the conflicting files.
* The clone link for the flexible environment.

> * The clone link for the mainline environment that flexible environment is connected to.

Reviewer 2: Conflicts can happen between any environment, that is merging together. 

Would it make sense to call them source and destination environment?

> * The clone link for the mainline (target) environment that flexible environment is connected to.

Reviewer: Are these interchanged? Isn't the mainline the source of the changes, and the flexible the target—since you're pulling from the mainline into flexible?

Another suggestion for clarity and consistency since the article mentions pushing/pulling terms: 

* The clone link for the mainline environment (where you're pulling changes from).
* The clone link for the flexible environment (where you're pulling changes into).

Author: Hm, good question!
The merge conflicts happens when you want to merge your changes from the Flex environment into the Mainline environment.
In this case, the Flex is the source and the mainline is the target environment.
You're pulling from the target first, in order to make sure it's in sync with the source environment, before deploying your changes to the target environment.

Does that make sense?
That's also how we're doing it throughout the Cloud docs, but usually there the "pulling" isn't mentioned, as this happens automatically when pushing from on environment to another 🤔

* An editor that provides Git tools for handling merge conflicts (like Visual Studio Code).

### Steps

1. Download the log file provided from the failed pull (see image above).
2. Clone the flexible environment [to your local machine](../../set-up/working-locally.md).

> 3. Use the mainline environment clone link to add a new `git remote:

Reviewer: ```suggestion
3. Use the mainline environment clone link to add a new `git remote`:
```
